### PR TITLE
TSA-136 migrate teams to container logic

### PR DESCRIPTION
**Pull request by alan-mj-lin**
- teams now rendered based on data queries
- server updates to get mobile app to login properly
- dragging and dropping for teams edit does not yet interact with the database to save changes

### CODE REVIEW

File: server/routes/tasks.ts [hunk 1]
```diff
@@ -27,18 +27,18 @@ router.get('/tasks', certify, async (req: Request, res: Response, next: NextFunc
...
 
     // }
-    if (churchId) {
-      const query = await db.Task.findAll({ where: searchParams });
-      tasks.push(...query);
-    }
+    // if (churchId) {
+    //   const query = await db.Task.findAll({ where: searchParams });
+    //   tasks.push(...query);
+    // }
```
Comment (hchiam) [inline server/routes/tasks.ts]: are we saving this for a later subtask?

File: server/routes/user-roles.ts [hunk 2]
```diff
@@ -38,6 +39,52 @@ router.get(
...
+            [Sequelize.Op.or]: roles.map((role) => role.id),
+          },
+        },
+        attributes: ['id', 'teamLead', 'roleId'],
+        include: [
+          { model: db.User, as: 'user', attributes: ['id', 'firstName', 'lastName'] },
+          { model: db.Role, as: 'role', attributes: ['name'] },
+        ],
+      });
+      console.log(userRoles[0].role.name);
```
Comment (hchiam) [inline server/routes/user-roles.ts]: Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) if it's not needed.

File: server/routes/user-roles.ts [hunk 3]
```diff
@@ -38,6 +39,52 @@ router.get(
...
+        userRoles.map((userRole) => {
+          if (role.name === userRole.role.name)
+            members.push({
+              id: userRole.user.id,
+              name: `${userRole.user.firstName} ${userRole.user.lastName}`,
+            });
+        });
+        if (members.length > 0) formattedData.push({ role: role.name, members: members });
+      });
+      console.log(formattedData);
```
Comment (hchiam) [inline server/routes/user-roles.ts]: Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) if it's not needed.

**Review by hchiam — CHANGES_REQUESTED** [review]
just `console.log`s, otherwise lgtm

File: server/routes/tasks.ts (hunk 1 above)
Comment (alan-mj-lin) [reply server/routes/tasks.ts]: someone confirm @showduhtung @CedThen that the Tasks table shouldn't have a churchId column. It was blocking the mobile app from login in so I commented it out. My version of the database doesn't have churchId in the Task table.

Comment (CedThen) [reply server/routes/tasks.ts]: yup, no churchId column in the tasks table

**Review by hchiam — APPROVED** [review]
LGTM! :+1:

File: src/components/Teams/DroppableTeamMemberList.tsx [hunk 4]
```diff
@@ -35,21 +37,28 @@ export const DroppableTeamMembersList = ({
...
+  const [addedMemberIds, setAddedMemberIds] = useState<string[]>(null);
+  const [deletedMemberIds, setDeletedMemberIds] = useState<string[]>(null);
 
   // handleSubmit and handleDelete are placeholder functions
   function handleSubmit(event: any) {
     event.preventDefault();
     if (members.find((member: any) => member.name === memberInput.value)) {
       setMemberInput({ ...memberInput, error: 'User already exists' });
+    } else if (!users.find((user: any) => user.name === memberInput.value)) {
+      setMemberInput({ ...memberInput, error: 'User does not exist' });
```
Comment (hchiam) [inline src/components/Teams/DroppableTeamMemberList.tsx:53]: why is one `members.find` and the other `!users.find`? superficially, i don't see why it's not `members` and `!members`, or `users` and `!users`.
Or is it clearer if the first error message is changed to `'User is already listed as a team member'`?

File: src/components/Teams/services.tsx [hunk 5]
```diff
@@ -38,18 +38,25 @@ export function add(
   droppableSource: DraggableLocation,
   droppableDestination: DraggableLocation,
 ): TeamState {
-  const stateClone: TeamData[] = teamsState[droppableDestination.droppableId];
+  const stateClone: TeamData[] = teamsState[droppableDestination.droppableId].members;
   const member = members[droppableSource.index];
+  console.log(member);
```
Comment (hchiam) [inline src/components/Teams/services.tsx:43]: Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) if it's not needed.

File: src/container/Teams/Teams.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  // useEffect(() => {
+  //   if (teams.isSuccess) setError(null);
+  //   if (teams.isError) setError(teams.error);
+
+  //   if (teams.isLoading !== isLoading) setIsLoading(teams.isLoading);
+  // }, [setUserData]);
+
+  console.log(userData, teams.data);
```
Comment (hchiam) [inline src/container/Teams/Teams.tsx:44]: Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) if it's not needed.

File: server/routes/tasks.ts [hunk 7]
```diff
@@ -14,8 +14,8 @@ router.get('/tasks', certify, async (req: Request, res: Response, next: NextFunc
     const searchArray = [];
     const { userId, churchId, roleId } = req.query;
     // is going to need more information like date, time --> events
-    if (userId) searchArray.push({ userId });
-    if (churchId) searchArray.push({ churchId });
+    if (userId) searchArray.push({ id: userId });
+    // if (churchId) searchArray.push({ churchId });
```
Comment (hchiam) [inline server/routes/tasks.ts:18]: `churchId` isn't used elsewhere in this scope. Are you leaving this line commented out just in case we change our mind?

**Review by hchiam — CHANGES_REQUESTED** [review]
It looks like you tested it, so just minor things from me.

**Pull request closed without merging**